jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ui/components/PreCodeTransformUserDialog.kt

```diff
@@ -71,49 +74,61 @@ class PreCodeTransformUserDialog(
-
         // Initialize model to hold form data
         val model = Model(
             focusedBuildFileIndex = focusedModuleIndex,
             focusedBuildFile = chosenBuildFile,
             focusedBuildFileModule = chosenModule,
-            selectedMigrationPath = supportedJavaVersions.firstOrNull(),
-            supportedMigrationPaths = supportedJavaVersions,
+            selectedJavaModuleVersion = tryToGetModuleJavaVersion(chosenModule),
+            targetUpgradeVersion = JavaSdkVersion.JDK_17,
         )
 
+        val jdkVersionLabel = JLabel(getJdkVersionText(model.selectedJavaModuleVersion))
         dialogPanel = panel {
             row { text(message("codemodernizer.customerselectiondialog.description.main")) }
             row { text(message("codemodernizer.customerselectiondialog.description.select")) }
             row {
-                buildFileComboBox = comboBox(buildfiles.map { it.path })
+                buildFileComboBox = comboBox(buildFiles.map { it.path })
                     .bind({ it.selectedIndex }, { t, v -> t.selectedIndex = v }, model::focusedBuildFileIndex.toMutableProperty())
                     .align(AlignX.FILL)
                     .columns(COLUMNS_MEDIUM)
                     .component
                 buildFileComboBox.whenItemSelected {
                     dialogPanel.apply() // apply user changes to model
-                    model.focusedBuildFile = buildfiles[model.focusedBuildFileIndex]
-                    model.focusedBuildFileModule = ModuleUtil.findModuleForFile(buildfiles[model.focusedBuildFileIndex], project)
-                    model.supportedMigrationPaths = supportedJdkForModuleOrProject(model.focusedBuildFileModule)
+                    model.focusedBuildFile = buildFiles[model.focusedBuildFileIndex]
+                    model.focusedBuildFileModule = ModuleUtil.findModuleForFile(buildFiles[model.focusedBuildFileIndex], project)
+                    model.selectedJavaModuleVersion = tryToGetModuleJavaVersion(model.focusedBuildFileModule)
                     dialogPanel.reset() // present model changes to user
+                    jdkVersionLabel.text = getJdkVersionText(model.selectedJavaModuleVersion)
+                    if (javaTransformInputSdks.contains(model.selectedJavaModuleVersion)) javaInputSdkComboBox.selectedItem = model.selectedJavaModuleVersion
                 }
                 buildFileComboBox.addActionListener {
                     CodetransformTelemetry.configurationFileSelectedChanged(
-                        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId()
+                        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     )
                 }
             }
+            row {
+                cell(jdkVersionLabel)
+            }
+            row {
+                javaInputSdkComboBox = comboBox(javaTransformInputSdks.map { it })
```

Comment:
Do we need to map the selectedIndex like the above combo box 
`.bind({ it.selectedIndex }, { t, v -> t.selectedIndex = v }, model::focusedJavaInputIndex.toMutableProperty())`?

Reply:
Ah yes thank you. Now I understand the purpose of that. Did much more thorough manual testing this time around to ensure everything works properly.